pyiceberg/utils/schema_conversion.py

```diff
@@ -524,12 +532,19 @@ def field(self, field: NestedField, field_result: AvroType) -> AvroType:
         if isinstance(field_result, dict) and field_result.get("type") == "record":
             field_result["name"] = f"r{field.field_id}"
 
+        orig_field_name = field.name
+        is_valid_field_name = _valid_avro_name(orig_field_name)
+        field_name = orig_field_name if is_valid_field_name else make_compatible_name(orig_field_name)
```

Comment:
nit: `make_compatible_name` already calls `is_valid_field_name` 
https://github.com/apache/iceberg-python/blob/904c0b77c33768716e8672b38bae73bfcf565fbf/pyiceberg/schema.py#L1357-L1361

we can simply this part
```suggestion
        field_name = make_compatible_name(orig_field_name)
```